### PR TITLE
#14267 Add endpoint to collect live-sensor data

### PR DESCRIPTION
**Pull request by azapasnik**

### CODE REVIEW

File: src/main/java/io/kontur/disasterninja/controller/FeaturesController.java [hunk 1]
```diff
@@ -26,15 +28,31 @@ public class FeaturesController {
...
     @ApiResponse(responseCode = "200", content = @Content(mediaType = APPLICATION_JSON_VALUE,
             array = @ArraySchema(schema = @Schema(implementation = FeatureDto.class))))
     @GetMapping
+    @Deprecated
     public List<FeatureDto> getUserAppFeatures(
             @RequestParam(name = "appId") @Parameter(name = "appId") UUID appId) {
         return userProfileClient.getUserAppFeatures(appId);
     }
+
+    @Operation(tags = "Features", summary = "Append user's live sensor data")
```
Comment (Komzpa) [inline src/main/java/io/kontur/disasterninja/controller/FeaturesController.java:45]: Can it be not live sensor data but just generic appending feature to the layer, with the rest of it handled on FE side?

File: src/main/java/io/kontur/disasterninja/service/LiveSensorFeatureService.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private final LayersApiClient layersApiClient;
+
+    public void appendLiveSensorData(FeatureCollection fc) {
+        String user = AuthenticationUtil.getAuthenticatedUser();
+        Feature[] features = Arrays.stream(fc.getFeatures())
+                .filter(f -> f.getProperties() != null && f.getGeometry() != null)
+                .peek(f -> f.getProperties().put("userId", user))
+                .toArray(Feature[]::new);
+
+        layersApiClient.appendLayerFeaturesWithDefaultUser("live-sensor", new FeatureCollection(features));
```
Comment (Komzpa) [inline src/main/java/io/kontur/disasterninja/service/LiveSensorFeatureService.java:25]: if this is layer name, can it be FE controlled?

**Review by Komzpa — CHANGES_REQUESTED** [review]
Can we make it a bit more generic please.

File: src/main/java/io/kontur/disasterninja/controller/FeaturesController.java (hunk 1 above)
Comment (azapasnik) [reply src/main/java/io/kontur/disasterninja/controller/FeaturesController.java:45]: I don't think so. This has quite specific logic handling userId and using default disaster.ninja user to append features into layer.
Layers-API would not let anyone to modify layers. Only layer owners can do. I've created live-sensor layer with disaster.ninja owner.

File: src/main/java/io/kontur/disasterninja/service/LiveSensorFeatureService.java (hunk 2 above)
Comment (azapasnik) [reply src/main/java/io/kontur/disasterninja/service/LiveSensorFeatureService.java:25]: I've specifically designed it to be feature specific to be able to deliver it fast and hide specific for this feature layer ownership transfer.

**Review by i1shilyaev — APPROVED** [review]
Apart Darafei's concerns regarding generic functionality LGTM

File: src/main/java/io/kontur/disasterninja/controller/FeaturesController.java (hunk 1 above)
Comment (Komzpa) [reply src/main/java/io/kontur/disasterninja/controller/FeaturesController.java:45]: It looks then that we need to backlog a ticket to redo the access model on the Layers API. Definitely not only the owner can edit the layers, but also people the owner opened some capabilities to. This particular use case shows that we have at least owner, logged-in, anonymous users kind and at least read, write, append capabilities. I believe that stamping the user into the feature can be done in a nicer way, like having a special data type for certain field in layer descriptor and just always stamping it on write if the spot to stamp the user at is present at the layer write.

**Review by Komzpa — APPROVED** [review]

**Pull request merged**